### ISSUE TITLE
if_switch_linter false positive when testing == ""

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#2835]
As seen in {DBItest}:

https://github.com/r-dbi/DBItest/blob/00060442200eb230350a308c37e30f797a371f95/R/spec-meta-bind-.R#L136-L153

```r
if (kind == "") {
  ...
} else if (kind == "1") {
  ...
} else if (kind == "name") {
  ...
} else {
  ...
}
```

This has no `switch()` equivalent because `""` would be a 0-length variable name.
